chain/client/src/stateless_validation/chunk_validator/mod.rs

```diff
@@ -95,30 +95,18 @@ impl ChunkValidator {
         let chunk_header = state_witness.chunk_header.clone();
         let network_sender = self.network_sender.clone();
         let epoch_manager = self.epoch_manager.clone();
-        if matches!(
-            pre_validation_result.main_transition_params,
-            chunk_validation::MainTransition::ShardLayoutChange
-        ) {
-            send_chunk_endorsement_to_block_producers(
-                &chunk_header,
-                epoch_manager.as_ref(),
-                signer,
-                &network_sender,
-            );
-            return Ok(());
-        }
 
-        // If we have the chunk extra for the previous block, we can validate the chunk without state witness.
+        // If we have the chunk extra for the previous block, we can validate
+        // the chunk without state witness.
         // This usually happens because we are a chunk producer and
         // therefore have the chunk extra for the previous block saved on disk.
         // We can also skip validating the chunk state witness in this case.
+        // We don't need to switch to parent shard uid, because resharding
+        // creates chunk extra for new shard uid.
```

Comment:
Just to double check, do the new shards' chunk extra contain the final state roots after applying the state changes on the parent and then splitting?

Reply:
Yes. For the last block hash in epoch, chunk extra for parent shard id includes results of applying last chunk, and there is also chunk extra for child shard id, which also includes split result on top of that.